Applications/ApplicationsLib/ProjectData.cpp

```diff
@@ -134,6 +147,44 @@ bool ProjectData::removeMesh(const std::string &name)
 	return mesh_found;
 }
 
+void ProjectData::buildProcesses()
+{
+	for (auto const& pc : _process_configs)
+	{
+		auto const type = pc.peekConfParam<std::string>("type");
+
+		auto const nl_slv_name = pc.getConfParam<std::string>("nonlinear_solver");
+		auto& nl_slv = BaseLib::get_or_error(_nonlinear_solvers, nl_slv_name,
+		    "A nonlinear solver with the given name has not been defined.");
+
+		auto time_disc = NumLib::createTimeDiscretization<GlobalVector>(
+		        pc.getConfSubtree("time_discretization")
+		    );
+
+		if (type == "GROUNDWATER_FLOW")
+		{
+			// The existence check of the in the configuration referenced
+			// process variables is checked in the physical process.
+			// TODO at the moment we have only one mesh, later there can be
+			// several meshes. Then we have to assign the referenced mesh
+			// here.
+			_processes.emplace_back(
+				ProcessLib::createGroundwaterFlowProcess<GlobalSetupType>(
+				    *_mesh_vec[0], *nl_slv, std::move(time_disc),
+				    _process_variables, _parameters, pc));
+		}
+		else
+		{
+			ERR("Unknown process type: %s", type.c_str());
+			std::abort();
+		}
+	}
+
+	// process configs are not needed anymore, so clear the storage
+	// in order to trigger config tree checks
+	_process_configs.clear();
```

Comment:
This line has not been introduced in this PR, rather it has been moved here from the header. Will fix it in a subsequent PR.